thrifty-schema/src/main/java/com/microsoft/thrifty/schema/Program.java

```diff
@@ -240,14 +242,52 @@ void loadIncludedPrograms(Loader loader, Set<Program> visited) throws IOExceptio
         this.includedPrograms = includes.build();
 
         LinkedHashMap<String, Named> symbolMap = new LinkedHashMap<>();
-        for (Named named : names()) {
-            Named oldValue = symbolMap.put(named.name(), named);
+
+        // Some type-resolution subtlety eludes me.  I'd have thought that
+        // Iterable<EnumType> is castable to Iterable<Named> (inheritance),
+        // but the IDE claims otherwise.  So, instead create a temp list where each list's
+        // data will be copied into
+        List<Named> namedList = new ArrayList<>();
```

Comment:
You're correct with the second point. I should have done that. I can make that change

I'm gonna think about the first issue you mentioned. Seems like the parser might have to get a little more complicated.